### PR TITLE
Base clojure template

### PR DESCRIPTION
**Pull request by marcitqualab**
Hello, this is the first version of the baseline clojure template.

### CODE REVIEW

**Comment by behrica** [comment]
Looks good.
There is of course no precise answer what "minimal" means.
Just 2 comments:
In this line you hardcode a version of `clj-new`..

https://github.com/scicloj/devcontainer-templates/blob/451edf34041056fd5b70b9bdf4f154e70b432fee/src/basecloj/.devcontainer/Dockerfile#L26

This means "conceptually", that each new version of `clj-new` requires an update of the template.
(If you do them or not, is then an other question)

Regarding `sshd`, I would prefer to remove it, as this is "security" related.
I would prefer not to have a default devcontainer config which adds an accessible ssh server.
So please remove this:
https://github.com/scicloj/devcontainer-templates/blob/451edf34041056fd5b70b9bdf4f154e70b432fee/src/basecloj/.devcontainer/devcontainer.json#L19

(does this related to https://github.com/scicloj/devcontainer-templates/blob/451edf34041056fd5b70b9bdf4f154e70b432fee/src/basecloj/.devcontainer/devcontainer.json#L12 ?
, if yes please remove as well)

File: src/basecloj/.devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+FROM clojure:temurin-21-tools-deps-jammy
```
Comment (behrica) [inline src/basecloj/.devcontainer/Dockerfile]: I think the "tag" of the image should be configurable.
Please see here:
https://github.com/scicloj/devcontainer-templates/blob/451edf34041056fd5b70b9bdf4f154e70b432fee/src/scicloj/devcontainer-template.json#L8

and

https://github.com/scicloj/devcontainer-templates/blob/c9ef0b703c0bd27f8883f5fcbf1937786f9aa995/src/scicloj/.devcontainer/Dockerfile#L1

File: src/basecloj/.devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// README at: https://github.com/scicloj/devcontainer-templates/tree/main/src/basecloj
+{
+	"name": "Base clojure dev env",
+	"build": {
+		"dockerfile": "Dockerfile",
+		"args": {
+			"USERNAME": "${localEnv:USER}"
+		}
+	},
+	"runArgs": [
```
Comment (behrica) [inline src/basecloj/.devcontainer/devcontainer.json]: plase remove this, if it is here due to sshd which should be removed

File: src/basecloj/.devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    # [Optional] Add sudo support. Omit if you don't need to install software after connecting.
+    && apt-get update \
+    && apt-get install -y sudo \
+    && echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME \
+    && chmod 0440 /etc/sudoers.d/$USERNAME
+    
+    
+# [Optional] Set the default user. Omit if you want to keep the default as root.
+USER $USERNAME
+SHELL ["/bin/bash", "-ec"]
```
Comment (behrica) [inline src/basecloj/.devcontainer/Dockerfile:20]: I tried as well to make "bash" the defaut shell, but did not succeed neither.When I open a "terminal" in VSCode, I get "sh" first and not "bash".

If you are aware of a way to get a "bash" by default, let me know.

**Comment by marcitqualab** [comment]
Hello @behrica , 

Yes the clj-new version for now is hardcoded. The official readme file recommends this way to install it. I have created a task on my side to research how to auto-update this version over time using the github actions. 

I have removed the ssh and security options.

Many thanks,

**Comment by behrica** [comment]
Changes looks good.
Do you still want to make the image tag configurable ?
as I commented on this file : src/basecloj/.devcontainer/Dockerfile

**Comment by marcitqualab** [comment]
Hello @behrica ,

I do not see any specific comment in the file : src/basecloj/.devcontainer/Dockerfile.
If you are referring to the clj-new version tag, I think that for now we can start with the current latest version as it is and I will research and test how to implement an auto update or use latest tag if possible.
I can create a new PR later on when I have an update on how to remove the hard coded clj-new version tag.  
Would that work for you?

**Comment by behrica** [comment]
No ,I talk about the "tag" of the base image
in this line in teh Dockerfile:
```
FROM clojure:temurin-21-tools-deps-jammy
```

you have hardcoded the tag of the image, namely:
`temurin-21-tools-deps-jammy`

But there are lots of others:
https://hub.docker.com/_/clojure/tags

And I think we should allow the user to "choose" when using the template.
This can be done via "template variables".
If we have those, VSCode renders the options and lets the user pick one,
the moment he is using the template.


The question is then:
- which options we make "choosable"
- whats the default ?


See here
https://github.com/scicloj/devcontainer-templates/blob/main/src/scicloj/devcontainer-template.json
what I think is the best solution for the "which options", (only LTS and JDK as supported by Clojure).
In any case a user can override it after applying the template.
But In VSCode he will only "see" the options we explicitly list.

**Comment by marcitqualab** [comment]
Ah, yes, got it. 
I will create some options then. 
Thanks.

**Review by behrica — APPROVED** [review]

**Pull request merged**